File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        

        n = len(nums)
        res = float("inf")
        L = 0
        cSum = 0
        # advance right pointer
        for R in range(n):
            # add to sum on right
            cSum += nums[R]
            # while meet condition
            while cSum >= target:
                # update min
                res = min(res,R-L+1)
                # advance left
                cSum -= nums[L]
                L+=1

        return 0 if res == float("inf") else res
            

File: Data Structures & Algorithms/concatenation-of-array/submission-2.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int N = nums.size();
        vector<int> res;
        res.resize(2*N);
        
        for (int i = 0; i < N; i++) {
            res[i] = res[i+N] = nums[i];
        }
        return res;
    }
};